frontend/lib/ui/ProgressBar/ProgressBar.tsx

```diff
@@ -2,28 +2,88 @@ import { cn } from "@kiesraad/util";
 
 import cls from "./ProgressBar.module.css";
 
+export type ProgressBarColorClass =
+  | "default"
+  | "errors_and_warnings"
+  | "unfinished"
+  | "in_progress"
+  | "first_entry_finished"
+  | "definitive"
+  | "not_started";
+
+export type RadiusClass = "no-radius" | "default-radius" | "radius-right" | "radius-left";
+export type PercentageAndColorClass = {
+  percentage: number;
+  class: ProgressBarColorClass;
+};
+
 export interface ProgressBarProps {
-  percent: number;
   id: string;
+  percentagesAndColorClasses: PercentageAndColorClass[];
   title?: string;
   spacing?: "small" | "large";
+  showPercentage?: boolean;
 }
-export function ProgressBar({ percent, id, title, spacing }: ProgressBarProps) {
+export function ProgressBar({
+  id,
+  percentagesAndColorClasses,
+  title,
+  spacing,
+  showPercentage = false,
+}: ProgressBarProps) {
+  const filteredBars = filterOutZeroPercentageBars(percentagesAndColorClasses);
+  const totalBars = filteredBars.length;
+  function filterOutZeroPercentageBars(bars: PercentageAndColorClass[]): PercentageAndColorClass[] {
+    return bars.filter((bar) => bar.percentage > 0);
+  }
+
+  function getRadiusClassForBarIndex(barIndex: number): RadiusClass {
+    if (totalBars === 1) {
+      return "default-radius";
+    }
+    if (barIndex === 0) {
+      return "radius-left";
+    } else if (barIndex === totalBars - 1) {
+      return "radius-right";
+    } else {
+      return "no-radius";
+    }
+  }
+
   return (
     <div className={cn(cls["progressbar-container"], spacing)} id={`progressbar-${id}`}>
       {title && <label>{title}</label>}
-      <section>
-        <div
-          className={cls["progressbar-outer"]}
-          role="progressbar"
-          aria-valuenow={percent}
-          aria-valuemin={0}
-          aria-valuemax={100}
-        >
-          <div className={cls["progressbar-inner"]} style={{ width: `${percent}%` }} />
-        </div>
-        <aside>{percent}%</aside>
-      </section>
+      {percentagesAndColorClasses.length === 1 && percentagesAndColorClasses[0] ? (
+        <section>
+          <div
+            className={cls["progressbar-outer"]}
+            role="progressbar"
+            aria-valuenow={percentagesAndColorClasses[0].percentage}
+            aria-valuemin={0}
+            aria-valuemax={100}
+          >
+            <div
+              className={cn(cls["progressbar-inner"], cls[percentagesAndColorClasses[0].class], cls["default-radius"])}
+              style={{ width: `${percentagesAndColorClasses[0].percentage}%` }}
+            />
+          </div>
+          {showPercentage && <aside>{percentagesAndColorClasses[0].percentage}%</aside>}
+        </section>
+      ) : (
+        <section>
+          <div className={cls["progressbar-outer"]} style={{ height: "0.75rem" }}>
```

Comment:
You could set a border-radius on this outer div, together with an overflow: hidden, and remove the border radii on the inner sections.

Reply:
Works like a charm, implemented in c3454c192dbc8c694a1c85fce5b8ed5ee45d7605